LibWeb: Handle special cases of PseudoElement::Type correctly

Totto16 opened the pull request:
Their are some special values for CSS::Selector::PseudoElement::Type which are after `KnownPseudoElementCount` and therefore not present in various arrays of pseudo elements, this leads to some errors, if a type after `KnownPseudoElementCount` is used without checking first. This adds explicit checks to all usages

Possible related PR #1362

Fixes #2904 

## Note

[dashdot](https://github.com/MauriceNino/dashdot) uses react and [simplebar-react](https://www.npmjs.com/package/simplebar-react) which internally does some compatibility checks for 

```js
getComputedStyle(el, "::-webkit-scrollbar").display
```

(where `el` is any element)
and that crashes, as per #2904

AtkinsSJ commented on Libraries/LibWeb/CSS/Selector.h:
```diff
@@ -68,7 +68,13 @@ class Selector : public RefCounted<Selector> {
+        }
+
+        static StringView
+        name(Selector::PseudoElement::Type pseudo_element);
```
You accidentally split this onto two lines, please remove that change.

AtkinsSJ commented on Libraries/LibWeb/Animations/Animatable.cpp:
```diff
@@ -187,14 +187,14 @@ void Animatable::visit_edges(JS::Cell::Visitor& visitor)
 GC::Ptr<CSS::CSSStyleDeclaration const> Animatable::cached_animation_name_source(Optional<CSS::Selector::PseudoElement::Type> pseudo_element) const
 {
-    if (pseudo_element.has_value())
+    if (pseudo_element.has_value() && CSS::Selector::PseudoElement::is_known_pseudo_element_type(pseudo_element.value()))
```
This change means that requesting the animation name source for a fake pseudo-elements will give you its element's one instead, which is wrong - it should give you nothing.

The same for the other places you've added this check. It's not correct to fall back to returning the element's data.

Totto16 replied:
That makes sense, that is not as easy, as there arise a few question on how to do that in the best way:
- The setters `set_cached_animation_name_animation` and `set_cached_animation_name_source`  need to be checked too, as far as I understand, or am I wrong?
- the getters return a `GC::Ptr`, is it okay to return `{}` (alias empty Ptr alias "nullptr")

I did this in a separate commit, so that it can be seen, what the changes were, I'll squash it, when it's correct 😄

AtkinsSJ reviewed (changes requested):
Please combine the first two commits: the second is a refactor of the first.

AtkinsSJ reviewed (commented):
This is looking good now I think! That last commit just needs squashing like you said.

Totto16 commented:
I squashed it now 👍🏼

Totto16 commented:
> Something went wrong with the squash - the test gets added in the first commit now, and removed in the second.

Yes, I just noticed, I used the git GUI and messed up, fixed it in the terminal now, 😓 😂 

It should be correct now
